tests/wlr_virtual_pointer_v1.cpp

```diff
@@ -235,10 +235,35 @@ TEST_F(VirtualPointerV1Test, when_virtual_pointer_scrolls_client_sees_axis)
     receive_client.roundtrip();
 }
 
-TEST_F(VirtualPointerV1Test, when_virtual_pointer_scrolls_with_steps_client_sees_axis_descrete)
+TEST_F(VirtualPointerV1Test, when_virtual_pointer_scrolls_with_steps_before_wl_pointer_v8_client_sees_only_axis_discrete)
 {
+    // Discrete scrolling is used before wl_pointer v8
+    if (receive_client.bind_if_supported(wl_seat_interface, wlcs::AtLeastVersion(8)))
+    {
+        GTEST_SKIP();
+    }
```

Comment:
I don't think `bind_if_supported()` can ever return null?

Reply:
It can't return null, but it will skip a test if it fails to bind. What I'm saying here is, "If you can successfully bind, then skip."

Reply:
Probably needs `wlcs::LessThanVersion(8)` (and some better support in `bind_if_supported()` for that scenario.)

Reply:
I now have a working version of `wlcs::LessThanVersion` working, but now that I think about it, I don't think this is exactly what we want in this case (though it may serve use in other cases.)

For this test, the only thing I want to know is the maximum version of an interface that a compositor supports. Something like `auto supports(wl_interface, VersionSpecifier) -> bool` sounds more in line of what we want. I specifically want to know that a compositor updated to wl_pointer v8 does not expect an `axis_discrete()` call.

What's most important, unless I'm thinking of this wrong, is the ability to test that WLCS cannot bind to a particular interface without the behavior of the test being automatically skipped.

Reply:
I'm not sure why we'd need to test that WLCS *can't* bind to a specific version?

If WLCS binds to v8 of the interface, it should get v8 behaviour. If we pass `LessThanVersion{8}` it should bind v7 (or earlier) and get v7 behaviour.